### PR TITLE
[wip]: add support for android TV

### LINKED ISSUE #5725: FR: Android TV
I know, I know. It is as stupid as it sounds. Who wants to run an e-reader app on a tv?.

Technically is possible and pretty easy: just detect if we run on a tv platform, set hasDpad=true and change orientation from "sensorPortrait" to "landscape".

I'm not sure if there's some hardcoded value somewhere (on the filemanager?) for height > width. Today tried to run an almost "vanilla" version on my new tv box and while the splash screen was in landscape mode once it reached the filemanager it turned to reverse portrait mode.

### PR DESCRIPTION
**Pull request by pazos**
Fixes #5725

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6327)

### CODE REVIEW

File: frontend/device/android/device.lua [hunk 1]
```diff
@@ -67,7 +67,9 @@ local Device = Generic:new{
     isAndroid = yes,
     model = android.prop.product,
     hasKeys = yes,
-    hasDPad = no,
+    hasDPad = function() return android.isTv() end,
+    hasFewKeys = function() return android.isTv() end,
```
Comment (Frenzie) [inline frontend/device/android/device.lua:71]: Aha! I knew it would pay off someday. :-)

(I'd think TV remotes had more buttons though?)

Comment (pazos) [reply frontend/device/android/device.lua:71]: The documentation says: https://developer.android.com/training/tv/start/controllers

So we have 6 buttons. (4 dpad, back and select). I use `hasFewKeys` as it opens the top menu in the reader with the right dpad button. Some things are still rough (ie: we don't need to close the menu with left dpad button, back is ok; there's no way to open top menu in the FM...)

**Comment by pazos** [comment]
Note to self:
- wait until the rotation is fixed for other devices.
- wait until a proper/native rotation is available on android
- implement device abstraction for Eink vs Lcd devices (https://github.com/koreader/koreader/issues/5768#issuecomment-652443499)
- implement android tv as a kind of Lcd device.

Finally figure out which is the best scheme to navigate the whole UI with dpad + back button.

File: frontend/device/android/device.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/device/android/device.lua:71]: I see. On Kindle etc. we have a menu key for the top menu and an ok/select/whatever key for the bottom menu. Just a menu key lacking is a lot less bad, but kind of annoying that it's not sufficiently full-featured.

> there's no way to open top menu in the FM

Perhaps something could be done with left and right, e.g., by reversing it from the normal FM button layout. Left for context menu, right for top menu.

Comment (pazos) [reply frontend/device/android/device.lua:71]: My suggestion would be:

## Reader
dpad-reader

## File manager
dpad-fm

## Menus
dpad-menu

Comment (Frenzie) [reply frontend/device/android/device.lua:71]: The reason I suggested right for top menu in the file manager is so it'd be consistent with the reader, or was that one for the bottom menu? We should try to keep it internally consistent with the reader too, within reason. Overall the above proposal looks good! :+1:

Comment (pazos) [reply frontend/device/android/device.lua:71]: Right now on the reader the center d-pad button opens the bottom menu. On the FM the right dpad button opens file menu.

Comment (Frenzie) [reply frontend/device/android/device.lua:71]: In that case I added → for the top menu (on hasFewKeys).

Comment (pazos) [reply frontend/device/android/device.lua:71]: That works on the reader, not on the filemanager, where it opens selected file/directory properties.

**Comment by pazos** [comment]
Everything works fine as is with a controller or a bluetooth keyboard. We're still missing keys with the (5dpad + back) remote that comes with most Tv boxes. I'll leave this as a perpetual **WIP** until we figure the right key combos. My proposal is in https://github.com/koreader/koreader/pull/6327#discussion_r450787837, but I don't want to implement it right now :P

**Comment by Commodore64user** [comment]
for the reader: if you simply add access to the bottom menu from the top one (like there is for the plus menu on the FM), you could free up the up/down keys. You could have:
left/right for page navigation
press or enter to call up the top menu (and bottom now too)
up/down are now free for other things like content selection on one and something else you value on the other.

for the menus: why not sacrifice the right and use that as long press?
up/down/left for navigation as you say
press or enter as select
right as long press

fm is about right that way.

doesn't that android documentation state that you have a menu button too? if that is the case you have even more leeway. otherwise consider my proposal.

**Pull request closed without merging**